packages/headless/src/controllers/url-manager/headless-url-manager.ts

```diff
@@ -77,25 +72,42 @@ export function buildUrlManager(
     throw loadReducerError;
   }
 
-  const {dispatch} = engine;
-  const controller = buildController(engine);
-
   validateInitialState(
     engine,
     initialStateSchema,
     props.initialState,
     'buildUrlManager'
   );
 
+  const completeParameters = (fragment: string) => ({
+    ...initialSearchParameterSelector(engine.state),
+    ...convertFragmentToParameters(fragment),
+  });
+
+  const controller = buildController(engine);
+  const {dispatch} = engine;
+  let previousFragment = decodeFragment(props.initialState.fragment);
```

Comment:
What would happen if someone instantiated multiple url manager controllers? Would there be any differences in the behavior of each if they have different previous fragments stored locally?

Reply:
It would probably function the same, all controller have a local state variable which they compare with new updates, but I don't the proposition as a valid use case, here's why:

Having 2 of these for an engine is currently useless, since we cannot specify which parameters to track, the 2 controllers would have the same exact same state, it would be identical as using 1 controller.

Now if we have a use case of multiple url managers, tracking a mix of different parameters (I don't really see a use case for that either really, you realistically have 1 url), I would see the fragments being saved in the state as sets, containing their config (params they're tracking) and their values.

Reply:
Yeah, as long as it does not "bug out" if you have a mistake in your code and you have multiple controller and it's just "useless", then that's 100% fine with me.

I don't see a use case either where you'd want to track different parameters on the same engine instance.

Reply:
Yep, as long as it works without bugs when multiple instances are created at different moments in the app life-cycle, then we should be good. Some developers might create multiple instances to change or set parameters from inside different components, so I wanted to make sure that the local state would not cause a problem.

Reply:
Creating multiple instances won't cause issues, but if you subscribe and sync to multiple instances simultaneously, you're going to end up changing the url more time than necessary, but the internals of the component won't bug out